orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/utils/TrafficGuard.java

```diff
@@ -164,6 +174,32 @@ private void verifyOtherServerGroupsAreTakingTraffic(String serverGroupName, Mon
     }
   }
 
+  private double getMinCapacityRatio() {
+    Double minCapacityRatio = dynamicConfigService.getConfig(Double.class, MIN_CAPACITY_RATIO, 0d);
+    if (minCapacityRatio == null || minCapacityRatio < 0 || 0.5 <= minCapacityRatio) {
+      log.error("Expecting a double value in range [0, 0.5) for {} but got {}", MIN_CAPACITY_RATIO, minCapacityRatio);
+      return 0;
+    }
+
+    return minCapacityRatio;
+  }
+
+  private List<Map> generateContext(List<TargetServerGroup> targetServerGroups) {
+    return targetServerGroups.stream()
+      .map(tsg -> ImmutableMap.builder()
+        .put("name", tsg.getName())
+        .put("disabled", tsg.isDisabled())
+        .put("instances", tsg.getInstances())
+        .build())
+      .collect(Collectors.toList());
+  }
+
+  private int getServerGroupCapacity(TargetServerGroup serverGroup) {
+    return (int) serverGroup.getInstances().stream()
+      .filter(instance -> "Up".equals(instance.get("healthState")))
```

Comment:
Did you consider using an asg's min or desired instead of service discovery health? During a production incident where many instances are unhealthy, min or desired may be preferable for calculation a capacity floor.

Reply:
I did not change this behavior, but I was surprised to find that this part of the code looked at number of instances actually up and not just the desired size for instance. They each have pros and cons, I'm not sure one number is the clear winner that should be used